application_test.go

```diff
@@ -277,23 +277,17 @@ func TestListApplication(t *testing.T) {
 	assert.Equal(t, len(application.Tasks), 2)
 }
 
-func TestHasApplication(t *testing.T) {
+func TestApplicationDoesntExists(t *testing.T) {
 	endpoint := newFakeMarathonEndpoint(t, nil)
 	defer endpoint.Close()
 
-	found, err := endpoint.Client.HasApplication(fakeAppName)
-	assert.NoError(t, err)
-	assert.True(t, found)
-
-	found, err = endpoint.Client.HasApplication("no_such_app")
-	assert.NoError(t, err)
-	assert.False(t, found)
+	_, err := endpoint.Client.Application("no_such_app")
+	assert.Equal(t, ErrDoesNotExist, err)
 
 	config := NewDefaultConfig()
 	config.URL = "http://non-existing-marathon-host.local:5555"
 	endpoint = newFakeMarathonEndpoint(t, &config)
 
-	found, err = endpoint.Client.HasApplication(fakeAppName)
+	_, err = endpoint.Client.Application(fakeAppName)
 	assert.Error(t, err)
```

Comment:
I'd also assert here that `err` does not equal `ErrDoesNotExist` in order to guard against the case that we might have created another case for a non-existing application unwittingly, either now or in the future.